### PR TITLE
T1150

### PR DESCRIPTION
**Pull request by humphd**
Add .jshintrc file for node jshint runner.

https://webmademovies.lighthouseapp.com/projects/65733/tickets/1150-better-jshint-global-jshint-flags

### CODE REVIEW

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Received

Command **cmd_check** from @jbuck received. Current queue size: 0

Live output at: http://butyr.org:8000/b250da9789c1d4c/output.txt

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Success

Full output at http://butyr.org:8000/b250da9789c1d4c/output.txt

Total script time: 0.29 mins

Checking 0e8301471f1865c1974251a3ce01080c848cd4d9 of pull request #285

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Received

Command **cmd_check** from @jbuck received. Current queue size: 0

Live output at: http://butyr.org:8000/68a02d15974db9c/output.txt

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Success

Full output at http://butyr.org:8000/68a02d15974db9c/output.txt

Total script time: 0.38 mins

Checking 0e8301471f1865c1974251a3ce01080c848cd4d9 of pull request #285

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Received

Command **cmd_check** from @jbuck received. Current queue size: 0

Live output at: http://butyr.org:8000/6e9bc3691f534e7/output.txt

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Success

Full output at http://butyr.org:8000/6e9bc3691f534e7/output.txt

Total script time: 0.27 mins

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Received

Command **cmd_check** from @jbuck received. Current queue size: 0

Live output at: http://butyr.org:8000/2278fe7b40c1c6f/output.txt

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Success

Full output at http://butyr.org:8000/2278fe7b40c1c6f/output.txt

Total script time: 0.24 mins

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Received

Command **cmd_preview** from @jbuck received. Current queue size: 0

Live output at: http://butyr.org:8000/38f53501e145dd9/output.txt

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Received

Command **cmd_check** from @jbuck received. Current queue size: 1

Live output at: http://butyr.org:8000/4553a924f8b2da3/output.txt

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Success

Full output at http://butyr.org:8000/38f53501e145dd9/output.txt

Total script time: 0.69 mins

Server is running on http://pr285.butyr.org/templates/test.html

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Failed

Full output at http://butyr.org:8000/4553a924f8b2da3/output.txt

Total script time: 0.01 mins

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Received

Command **cmd_check** from @jbuck received. Current queue size: 0

Live output at: http://butyr.org:8000/307aca6929ed939/output.txt

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Failed

Full output at http://butyr.org:8000/307aca6929ed939/output.txt

Total script time: 0.00 mins

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Received

Command **cmd_check** from @jbuck received. Current queue size: 0

Live output at: http://butyr.org:8000/32818ebb7a5d984/output.txt

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Failed

Full output at http://butyr.org:8000/32818ebb7a5d984/output.txt

Total script time: 0.57 mins

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Received

Command **cmd_check** from @jbuck received. Current queue size: 0

Live output at: http://butyr.org:8000/37486ea682f9452/output.txt

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Failed

Full output at http://butyr.org:8000/37486ea682f9452/output.txt

Total script time: 0.21 mins

File: .jshintrc [hunk 1]
```diff
@@ -0,0 +1,31 @@
+{
+    "predef": [
```
Comment (jbuck) [inline .jshintrc]: I don't see this listed on the jshint options page, what does it do? http://www.jshint.com/options/

File: .jshintrc [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+        "Popcorn",
+        "requirejs",
+        "require",
+        "define",
+        "Range",
+        "DOMException",
+        "console"
+    ],
+
+    "node" : true,
```
Comment (jbuck) [inline .jshintrc]: Can we order all of these options in the same order that http://www.jshint.com/options/ does? So it's alphabetical order, split into 3 sections: Enforcing Options, Relaxing options, and environment options.

File: .jshintrc [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+    "es5" : true,
+    "browser" : true,
+    "devel": true,
+
+    "indent": 2,
```
Comment (jbuck) [inline .jshintrc]: this looks like it doens't exist either

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Received

Command **cmd_check** from @jbuck received. Current queue size: 0

Live output at: http://butyr.org:8000/8d5e39846edb314/output.txt

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Failed

Full output at http://butyr.org:8000/8d5e39846edb314/output.txt

Total script time: 0.37 mins

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Received

Command **cmd_check** from @jbuck received. Current queue size: 0

Live output at: http://butyr.org:8000/887249a40419f1c/output.txt

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Success

Full output at http://butyr.org:8000/887249a40419f1c/output.txt

Total script time: 0.31 mins

File: src/plugin/module.js [hunk 4]
```diff
@@ -212,9 +212,9 @@
               for( var i=0; i<_listContainer.childNodes.length; ++i ){
                 if( _listContainer.childNodes[ i ].getAttribute( "data-butter-plugin-type" ) === pluginType ){
                   return;
-                }  
+                }
               }
-              _listContainer.appendChild( plugin.createElement( _pattern ) );  
+              _listContainer.appendChild( plugin.createElement( _pattern, butter ) );
```
Comment (jbuck) [inline src/plugin/module.js]: Change the argument order to `butter, _pattern`

File: src/plugin/plugin.js [hunk 5]
```diff
@@ -182,9 +184,7 @@ define( [ "util/dragndrop" ], function( DragNDrop ){
       }
     });
 
-    _helper = document.getElementById( _this.type + "-icon" ) || document.getElementById( "default-icon" );
-
-    this.createElement = function ( pattern ) {
+    this.createElement = function ( pattern, butter ) {
```
Comment (jbuck) [inline src/plugin/plugin.js]: Change the argument order to `butter, _pattern`

File: src/timeline/track-container.js [hunk 6]
```diff
@@ -12,7 +12,7 @@ function(
   DragNDrop
 ) {
 
-  return function( media ){
+  return function( media, butter ){
```
Comment (jbuck) [inline src/timeline/track-container.js]: Change the argument order to `butter, media`

File: src/timeline/media.js [hunk 7]
```diff
@@ -28,9 +28,22 @@ define( [
...
+        _zoom = nextZoom;
+        _tracksContainer.zoom = _zoom;
+        updateUI();
+      } //if
+    } //zoomCallback
+
     var _this = this,
         _media = media,
-        _tracksContainer = new TrackContainer( media ),
+        _tracksContainer = new TrackContainer( media, butter ),
```
Comment (jbuck) [inline src/timeline/media.js]: Change the argument order to `butter, media`

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Received

Command **cmd_cleanup** from @jbuck received. Current queue size: 0

Live output at: http://butyr.org:8000/04038b3c226114c/output.txt

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Failed

Full output at http://butyr.org:8000/04038b3c226114c/output.txt

Total script time: 1.33 mins

**Pull request merged**